sky/packages/sky/lib/src/fn3/focus.dart

```diff
@@ -137,8 +140,9 @@ class FocusState extends State<Focus> {
   void _scopeRemoved(GlobalKey key) {
     assert(_focusedScope == key);
     _currentlyRegisteredScopeRemovalListenerKey = null;
-    setState(() {
-      _focusedScope = null;
+    _focusedScope = null;
```

Comment:
How can it do that?  At the time we call this function (after exiting the last buildScope) we're not going to build again until the next frame.